test/rules/no-mocha-arrows.js

```diff
@@ -16,35 +16,73 @@ ruleTester.run('no-mocha-arrows', rules['no-mocha-arrows'], {
             code: 'function it () {}; it(() => { console.log("okay") })',
             parserOptions: { ecmaVersion: 6 }
         },
-        'it.only()'
+        'it.only()',
+        'it(function(done) { assert(something, false); done(); })',
+        {
+            code: 'it(function*() { assert(something, false) })',
+            parserOptions: { ecmaVersion: 6 }
+        },
+        {
+            code: 'it(async function () { assert(something, false) })',
+            parserOptions: { ecmaVersion: 2017 }
+        }
     ],
 
     invalid: [
         {
             code: 'it(() => { assert(something, false); })',
             parserOptions: { ecmaVersion: 6 },
-            errors: [ { message: expectedErrorMessage, column: 1, line: 1 } ]
+            errors: [ { message: expectedErrorMessage, column: 1, line: 1 } ],
```

Comment:
I usually extract the common errors into a variable and just put it everywhere. Removes a lot of duplicated code. but @lo1tuma may not have the same preference.

```js
const errors =  [ { message: 'Do not pass arrow functions to it()', column: 1, line: 1 } ];

 ruleTester.run('no-mocha-arrows', rules['no-mocha-arrows'], {
 // ...
  {
    code: '...',
    errors: errors
  }
});
```